google/generativeai/types/content_types.py

```diff
@@ -240,15 +249,360 @@ def to_contents(contents: ContentsType) -> list[glm.Content]:
     return contents
 
 
-ToolsType = Union[Iterable[glm.Tool], glm.Tool, dict[str, Any], None]
+def _generate_schema(
+    f: Callable[..., Any],
+    *,
+    descriptions: Mapping[str, str] = {},
+    required: Sequence[str] = [],
```

Comment:
Just realised that these default values are mutable.

Reply:
!